addons/binding/org.openhab.binding.milight/src/main/java/org/openhab/binding/milight/internal/protocol/MilightDiscover.java

```diff
@@ -77,61 +86,59 @@ public MilightDiscover(DiscoverResult discoverResult, int resendTimeoutInMillis,
         }
         interrupt();
-        if (datagramSocket != null) {
-            datagramSocket.close();
-            datagramSocket = null;
-        }
     }
 
     /**
-     * Used by the scheduler to resend discover messages. Stops after 3 attempts.
+     * Used by the scheduler to resend discover messages. Stops after a configured amount of attempts.
      */
     private class SendDiscoverRunnable implements Runnable {
         @Override
         public void run() {
+            // Stop after a certain amount of attempts
             if (++resendCounter > resendAttempts) {
-                if (resendTimer != null) {
-                    resendTimer.cancel(false);
-                    resendTimer = null;
+                stopResend();
+                // If we tried to discover a specific bridge, we apparently failed. Report this to the observer.
+                if (destIP != null) {
+                    discoverResult.noBridgeDetected();
                 }
-                discoverResult.noBridgeDetected();
                 return;
             }
 
             if (destIP != null) {
                 sendDiscover(destIP);
-            } else {
-                Enumeration<NetworkInterface> e;
-                try {
-                    e = NetworkInterface.getNetworkInterfaces();
-                } catch (SocketException e1) {
-                    logger.error("Could not enumerate network interfaces for sending the discover packet!");
-                    stopResend();
-                    return;
-                }
-                while (e.hasMoreElements()) {
-                    NetworkInterface networkInterface = e.nextElement();
-                    Iterator<InterfaceAddress> it = networkInterface.getInterfaceAddresses().iterator();
-                    while (it.hasNext()) {
-                        InterfaceAddress address = it.next();
-                        if (address == null) {
-                            continue;
-                        }
-                        InetAddress broadcast = address.getBroadcast();
-                        if (broadcast != null && !address.getAddress().isLoopbackAddress()) {
-                            sendDiscover(broadcast);
-                        }
+                return;
+            }
+
+            Enumeration<NetworkInterface> e;
+            try {
+                e = NetworkInterface.getNetworkInterfaces();
+            } catch (SocketException e1) {
+                logger.error("Could not enumerate network interfaces for sending the discover packet!");
+                stopResend();
+                return;
+            }
+            while (e.hasMoreElements()) {
+                NetworkInterface networkInterface = e.nextElement();
+                Iterator<InterfaceAddress> it = networkInterface.getInterfaceAddresses().iterator();
+                while (it.hasNext()) {
+                    InterfaceAddress address = it.next();
+                    if (address == null) {
```

Comment:
Majority does of examples using this code does not do null checks but that is not a guarantee ofcourse: http://www.programcreek.com/java-api-examples/index.php?class=java.net.NetworkInterface&method=getInterfaceAddresses